target/linux/ipq40xx/image/generic.mk

```diff
@@ -1044,19 +1044,30 @@ define Device/unielec_u4019-32m
 endef
 TARGET_DEVICES += unielec_u4019-32m
 
-define Device/zte_mf286d
+define Device/zte_mf28x_common
 	$(call Device/FitzImage)
 	DEVICE_VENDOR := ZTE
-	DEVICE_MODEL := MF286D
 	SOC := qcom-ipq4019
 	DEVICE_DTS_CONFIG := config@ap.dk04.1-c1
 	BLOCKSIZE := 128k
 	PAGESIZE := 2048
 	KERNEL_IN_UBI := 1
 	DEVICE_PACKAGES := kmod-usb-net-qmi-wwan kmod-usb-serial-option uqmi
 endef
+
+define Device/zte_mf286d
+	$(call Device/zte_mf28x_common)
+	DEVICE_MODEL := MF286D
+endef
 TARGET_DEVICES += zte_mf286d
 
+define Device/zte_mf289f
+	$(call Device/zte_mf28x_common)
+	DEVICE_MODEL := MF289F
+	DEVICE_PACKAGES := ipq-wifi-zte_mf289f ath10k-firmware-qca9984-ct
```

Comment:
I think you meant `DEVICE_PACKAGES += ipq-wifi-zte_mf289f ath10k-firmware-qca9984-ct`, otherwise this will overwrite `DEVICE_PACKAGES` inherited from `Device/zte_mf28x_common`

Reply:
done :)